src/widget/translationAnimator.ts

```diff
@@ -169,11 +161,51 @@ export class TranslationAnimator extends Clutter.Actor {
         this.transitionContainer.ease(transitionConfig);
     }
 
+    setActors(actors: Clutter.Actor[]) {
+        this.transitionContainer.remove_all_children();
+        for (const actor of actors) {
+            reparentActor(actor, this.transitionContainer);
+        }
+        this.currentActors = actors;
+    }
+
     endTransition(): void {
         this.transitionContainer.translation_x = 0;
         this.transitionContainer.translation_y = 0;
         this.animationInProgress = false;
         this.emit('transition-completed');
-        this.transitionContainer.remove_all_children();
+        for (const child of this.transitionContainer.get_children()) {
+            if (!this.currentActors.includes(child)) {
+                this.transitionContainer.remove_actor(child);
+            }
+        }
+    }
+
+    override vfunc_get_preferred_height(
+        _forWidth: number
+    ): [number | null, number | null] {
+        const parent = this.get_parent();
+        if (parent) {
+            return [
+                parent.allocation.get_height(),
+                parent.allocation.get_height(),
```

Comment:
Is this proper? `vfunc_get_preferred_height` shouldn't depend on its parent's allocation? That seems like a very cyclical dependency.

Reply:
For the animation purpose we need it to have the size of the parent and we used to manually assigned it and I thought it was better to have something dynamic but yeah it doesn't seem to be good dependencies.
It's could be better to change the parent allocation

Reply:
Yeah. I'm not sure which object is the parent of this, but that one is responsible for setting the allocation.
Can't the translation animator just signify 100% height/100% width?